gobblin-data-management/src/main/java/org/apache/gobblin/data/management/conversion/hive/task/HiveMaterializerQueryGenerator.java

```diff
@@ -0,0 +1,228 @@
+    this.conversionConfig = hiveDataset.getConversionConfigForFormat("sameAsSource").get();
+    this.outputDatabaseName = conversionConfig.getDestinationDbName();
+    this.outputTableName = conversionConfig.getDestinationTableName();
+    this.outputDataLocation = HiveConverterUtils.getOutputDataLocation(conversionConfig.getDestinationDataPath());
+    this.outputStagingTableName = HiveConverterUtils.getStagingTableName(conversionConfig.getDestinationStagingTableName());
+    this.outputStagingDataLocation = HiveConverterUtils.getStagingDataLocation(conversionConfig.getDestinationDataPath(), outputStagingTableName);
+    this.sourceDataPathIdentifier = conversionConfig.getSourceDataPathIdentifier();
+    this.pool = HiveMetastoreClientPool.get(workUnitState.getJobState().getProperties(),
+        Optional.fromNullable(workUnitState.getJobState().getProp(HiveDatasetFinder.HIVE_METASTORE_URI_KEY)));
+    this.conversionEntity = getConversionEntity();
+    this.stagingDataPartitionDirName = HiveConverterUtils.getStagingDataPartitionDirName(conversionEntity, sourceDataPathIdentifier);
+    this.stagingDataPartitionLocation = outputStagingDataLocation + Path.SEPARATOR + stagingDataPartitionDirName;
+    this.partitionsDDLInfo = Maps.newHashMap();
+    this.partitionsDMLInfo = Maps.newHashMap();
+    HiveConverterUtils.populatePartitionInfo(conversionEntity, partitionsDDLInfo, partitionsDMLInfo);
+    this.destinationTableMeta = HiveConverterUtils.getDestinationTableMeta(outputDatabaseName,
+        outputTableName, workUnitState).getLeft();
+  }
+
+  @Override
+  public List<String> generateQueries() {
+
+    List<String> hiveQueries = Lists.newArrayList();
+
+    Preconditions.checkNotNull(this.workUnit, "Workunit must not be null");
+    EventWorkunitUtils.setBeginDDLBuildTimeMetadata(this.workUnit, System.currentTimeMillis());
+
+    HiveConverterUtils.createStagingDirectory(fs, conversionConfig.getDestinationDataPath(),
+        conversionEntity, this.workUnitState);
+
+    // Create DDL statement for table
+    String createStagingTableDDL =
+        HiveConverterUtils.generateCreateDuplicateTableDDL(
+            inputDbName,
+            inputTableName,
+            outputStagingTableName,
+            outputStagingDataLocation,
+            Optional.of(outputDatabaseName));
+    hiveQueries.add(createStagingTableDDL);
+    log.debug("Create staging table DDL:\n" + createStagingTableDDL);
+
+    // Create DDL statement for partition
+    if (partitionsDMLInfo.size() > 0) {
+      List<String> createStagingPartitionDDL =
+          HiveAvroORCQueryGenerator.generateCreatePartitionDDL(outputDatabaseName,
+              outputStagingTableName,
+              stagingDataPartitionLocation,
+              partitionsDMLInfo);
+
+      hiveQueries.addAll(createStagingPartitionDDL);
+      log.debug("Create staging partition DDL: " + createStagingPartitionDDL);
+    }
+
+    hiveQueries.add("SET hive.exec.dynamic.partition.mode=nonstrict");
```

Comment:
It is improving readability of the code.
Query is "insert into target_table partition('partition_name') select * from source table" 
If we do not set dynamic partition mode to nonstrict, we will have to write partition values also, and because hive considers partition as a virtual column, we also have to write each of the column name in the query (in place of *) to match source and target columns.